plugins/by-name/alpha/default.nix

```diff
@@ -156,20 +165,27 @@ in
       opt = options.plugins.alpha;
     in
     lib.mkIf cfg.enable {
-      # TODO: deprecated 2024-08-29 remove after 24.11
-      warnings = lib.mkIf opt.iconsEnabled.isDefined [
-        ''
-          nixvim (plugins.alpha):
-          The option definition `plugins.alpha.iconsEnabled' in ${lib.showFiles opt.iconsEnabled.files} has been deprecated; please remove it.
-          You should use `plugins.alpha.iconsPackage' instead.
-        ''
-      ];
 
-      extraPlugins =
-        [ cfg.package ]
-        ++ lib.optional (
-          cfg.iconsPackage != null && (opt.iconsEnabled.isDefined -> cfg.iconsEnabled)
-        ) cfg.iconsPackage;
+      # TODO: added 2024-09-20 remove after 24.11
+      warnings =
+        lib.optional
+          (
+            opt.iconsEnabled.isDefined
+            || (
+              (opt.iconsEnabled.isDefined -> cfg.iconsEnabled)
+              && options.plugins.web-devicons.enable.highestPrio == 1490
+            )
```

Comment:
This is logically redundant. Both the LHS and RHS of the `||` require `isDefined`. Therefore the RHS has no effect.

I guess what we want is _either_:
- show one warning when the RHS is _not_ true and a different warning when it _is_ true
- have a single warning, but dynamically adapt the warning text depending on whether the RHS is true.

Reply:
E.g.

```nix
lib.optional opt.iconsEnabled.isDefined 
   (
     if cfg.iconsEnabled && options.plugins.web-devicons.enable.highestPrio == 1490 then
      "warning 1"
    else
      "warning 2"
   )
```

Reply:
I understand that they are separate, this approach was in response to :

>Yeah, ideally we'd only print one warning message. I don't mind if it's one "dumb" all encompassing messages, or if it takes additional conditions into account, but we should definitely always have some warning when the option is defined.

I was just making sure we print the same warning when the option is used, at all. I didn't care if the user had used it differently, I'm just letting them know we don't have the option available anymore and we will print the warning telling them to use a new option.

Reply:
If we're going back to wanting two separate warnings I would just revert it back to how I had it originally.